src/kinetics/Kinetics.cpp

```diff
@@ -755,14 +712,6 @@ void Kinetics::modifyReaction(size_t i, shared_ptr<Reaction> rNew)
             rOld->type(), rNew->type());
     }
 
-    if (!(rNew->usesLegacy())) {
-        if (rNew->rate()->type() != rOld->rate()->type()) {
-            throw CanteraError("Kinetics::modifyReaction",
-                "ReactionRate types are different: {} != {}.",
-                rOld->rate()->type(), rNew->rate()->type());
-        }
-    }
```

Comment:
Shouldn't this be retained, but always take place now? I didn't think switching the rate parameterization worked.

Reply:
Good catch!